CorsixTH/Lua/entities/patient.lua

```diff
@@ -245,16 +249,17 @@ function Patient:treated() -- If a drug was used we also need to pay for this
       if self.is_emergency then
         self.hospital.emergency.cured_emergency_patients = hospital.emergency.cured_emergency_patients + 1
       end
+
       self.cured = true
       self.infected = false
-      self:setMood("cured", "activate")
-      self.world.ui:playSound "cheer.wav" -- This sound is always heard
       self.attributes["health"] = 1
-      self:changeAttribute("happiness", 0.8)
-      hospital:changeReputation("cured", self.disease)
-      self.treatment_history[#self.treatment_history + 1] = _S.dynamic_info.patient.actions.cured
-      self:goHome(self.cured)
       self:updateDynamicInfo(_S.dynamic_info.patient.actions.cured)
+      self.hospital:msgCured()
+
+      -- If the patient refused to pay they are on the way home already
+      if not self.going_home then
```

Comment:
Looks like `dealtWithPatient()` would be impacted as well (it calls `receiveMoneyForTreatment()`); so we could also possibly send the patients home from there?
It's actually quite a big change, I'll have a look at it later and see if we can clarify the workflow and the encapsulation.